rocAL/rocAL/source/sequence_file_source_reader.cpp

```diff
@@ -0,0 +1,292 @@
+/*
+Copyright (c) 2019 - 2020 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <cassert>
+#include <algorithm>
+#include <commons.h>
+#include "sequence_file_source_reader.h"
+#include <boost/filesystem.hpp>
+
+namespace filesys = boost::filesystem;
+
+SequenceFileSourceReader::SequenceFileSourceReader() : _shuffle_time("shuffle_time", DBG_TIMING)
+{
+    _src_dir = nullptr;
+    _sub_dir = nullptr;
+    _entity = nullptr;
+    _curr_file_idx = 0;
+    _current_file_size = 0;
+    _current_fPtr = nullptr;
+    _loop = false;
+    _sequence_id = 0;
+    _shuffle = false;
+    _sequence_count_all_shards = 0;
+}
+
+unsigned SequenceFileSourceReader::count()
+{
+    if (_loop)
+        return _frame_names.size();
+    int ret = ((int)_frame_names.size() - _read_counter);
+    return ((ret < 0) ? 0 : ret);
+}
+
+Reader::Status SequenceFileSourceReader::initialize(ReaderConfig desc)
+{
+    auto ret = Reader::Status::OK;
+    _sequence_id = 0;
+    _folder_path = desc.path();
+    _shard_id = desc.get_shard_id();
+    _shard_count = desc.get_shard_count();
+    _user_batch_count = desc.get_batch_size();
+    _shuffle = desc.shuffle();
+    _loop = desc.loop();
+    _sequence_length = desc.get_sequence_length();
+    _step = desc.get_frame_step();
+    _stride = desc.get_frame_stride();
+    _batch_count = _user_batch_count / _sequence_length;
+    ret = subfolder_reading();
+    if (ret != Reader::Status::OK)
+        return ret;
+    ret = get_sequences();
```

Comment:
don't you need to check the status in the following code as well?

Reply:
The status has been checked in line number#86